src/Docker.PowerShell/Cmdlets/DockerFactory.cs

```diff
@@ -26,20 +29,52 @@ public class DockerFactory
     public static DockerClient CreateClient(string HostAddress, string CertificateLocation)
     {
         HostAddress = HostAddress ?? Environment.GetEnvironmentVariable("DOCKER_HOST") ?? DefaultHost;
-                    
+
         CertificateLocation = CertificateLocation ?? Environment.GetEnvironmentVariable("DOCKER_CERT_PATH");
 
         CertificateCredentials cred = null;
         if (!string.IsNullOrEmpty(CertificateLocation))
         {
+#if !NET46
+            throw new InvalidOperationException("TLS authetication is not supported in .NET Core.");
+#else
             // Try to find a certificate for secure connections.
             cred = new CertificateCredentials(
-                    new X509Certificate2(
-                        System.IO.Path.Combine(CertificateLocation, KeyFileName),
-                        certPass));
+                RSAUtil.GetCertFromPEMFiles(
```

Comment:
Is there any reason at all to check if the extension is pfx or pem in CertificateLocation? IE: Should we support either option?